src/source/source_cache.js

```diff
@@ -573,10 +591,12 @@ class SourceCache extends Evented {
         if (tile.uses > 0)
             return;
 
-        tile.stopPlacementThrottler();
+        this._updatePlacement();
+        tile.removed();
 
         if (tile.hasData()) {
-            const wrappedId = tile.coord.wrapped().id;
+            tile.coord = tile.coord.wrapped();
```

Comment:
When does this change the value of `tile.coord`? Does this affect anything else later on?

Reply:
The tile is being added to the cache with the wrapped ID. If we don't modify the tile's ID to match the wrapped ID, then when the tile is pulled out of the cache, the ID stored in the tile won't match the ID that was used to fetch it. This messes up the `CrossTileSymbolIndex` logic, which is based on the ID stored in the tile.

The sequence is something like this (I'm going from memory here). Imagine tile A and its wrapped copy tile B.

 - Tile B added
 - Tile B removed (now in cache with Tile A's id)
 - Tile A added (fetches tile B from cache). CrossTileSymbolIndex adds entries for Tile B.
 - Tile B added. CrossTileSymbolIndex doesn't do anything because tile B is already added.
 - Tile A removed. CrossTileSymbolIndex removes entries for Tile B, even though they're still relevant for Tile B. Unpredictable label madness ensues.